scheduler/src/cook/mesos/constraints.clj

```diff
@@ -211,12 +218,18 @@
    sequence of Fenzo TaskTracker.ActiveTasks, which correspond to ALL the running tasks in group.
    This includes both running tasks and tasks that are being assigned by Fenzo
    in the current cycle."
-  [db group task-id same-cycle-task-ids task-tracker-state]
-  (let [cotask-ids (disj
-                     (clojure.set/union
-                       (group/group->running-task-id-set db group) ; Tasks that are in the database (running or scheduled)
-                       same-cycle-task-ids) ; Tasks in same cycle
-                     task-id)] ; Do not include this task-id
+  [db group task-id same-cycle-task-ids task-tracker-state running-task-ids-cache]
+  (let [group-id (:group/uuid group)
+        db-cotask-ids (cache/lookup (swap! running-task-ids-cache
+                                           (fn [c]
+                                             (if (cache/has? c group-id)
+                                               (cache/hit c group-id)
+                                               (cache/miss c group-id (group/group->running-task-id-set db group)))))
+                                    group-id)
+        cotask-ids (disj (clojure.set/union
+                          db-cotask-ids ; Tasks that are in the database (running or scheduled)
+                          same-cycle-task-ids) ; Tasks in same cycle
+                         task-id)] ; Do not include this task-id
     ; Return a Fenzo TaskTracker.ActiveTask
     (get-cotasks-from-tracker-state task-tracker-state cotask-ids)))
```

Comment:
On my inital read this looked like a faulty recursive call to me. The function name is `get-cotasks-from-tracker-state`, but the arguments are different. Can we rename one of these functions?

Reply:
Oops, my bad. The code collapsed enough to hide the declaration of `get-fenzo-cotasks`